### PR TITLE
Add range support

### PR DESCRIPTION
**Pull request by nikitagromov**
Pivotal story id [#131987231](https://www.pivotaltracker.com/story/show/131987231)

### CODE REVIEW

**Comment by nikitagromov** [comment]
Need to be merged after [this PR](https://github.com/geniusproject/nefertari/pull/57)

File: nefertari/es_query.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+            for value in self.values:
+                if isinstance(value, Node):
+                    strings.append(value.parse())
+                else:
+                    strings.append(value)
+            return strings
+
+    head = Node()
+
+    for index, token in enumerate(tokens):
```
Comment (nmenardg) [inline nefertari/es_query.py]: `enumerate` is not necessary here

File: nefertari/es_query.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+            continue
+        if token == ')':
+            head = head.prev
+            continue
+
+        head.values.append(token)
+    return head.parse()
+
+
+def _remove_needles_brackets(tokens):
```
Comment (nmenardg) [inline nefertari/es_query.py]: _remove_needless_parentheses\*

File: nefertari/es_query.py [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+    split string in first matching with key
+    :param item: string which contain field_name:value or field_name:[00:00:00 TO 01:00:00]
+    :param split_key: key, which we use to split string
+    :return:
+    """
+    split_index = -1
+    for index, key in enumerate(item):
+        if key == split_key:
+            split_index = index
+            break
```
Comment (nmenardg) [inline nefertari/es_query.py]: This is essentially what str.find() does:
https://docs.python.org/3.4/library/stdtypes.html#str.find

File: tests/test_es_query.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        query_string = '((item:value OR item:value) AND NOT (item:OR OR NOT item:NOT)) OR true:true'
+        assert _get_tokens(query_string) == ['(', '(', 'item:value', 'OR', 'item:value', ')',
+                                             'AND NOT', '(', 'item:OR', 'OR NOT', 'item:NOT',
+                                             ')', ')', 'OR', 'true:true']
+
+    def test_remove_top_level_brackets_if_needless(self):
+        query_string = '(item:value OR item:value)'
+        tokens = _get_tokens(query_string)
+        tree = _build_tree(tokens)
+        assert ['item:value', 'OR', 'item:value'] == tree
```
Comment (nmenardg) [inline tests/test_es_query.py]: Please follow our convention for equalities:
`actual == expected`
In this case, `tree` is actual, `['item:value', 'OR', 'item:value']` is expected

File: tests/test_es_query.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+    def test_build_tree(self):
+        query_string = '(item:value OR item:value) AND ((item:value OR item:value AND complicated:false) OR (item:value OR item:value))'
+        tokens = _get_tokens(query_string)
+        tree = _build_tree(tokens)
+        assert [['item:value', 'OR', 'item:value'], 'AND',
+                [['item:value', 'OR', 'item:value', 'AND', 'complicated:false'], 'OR',
+                 ['item:value', 'OR', 'item:value']]] == tree
```
Comment (nmenardg) [inline tests/test_es_query.py]: `actual == expected`

**Review by nmenardg — CHANGES_REQUESTED** [review]

**Review by nmenardg — APPROVED** [review]

**Pull request merged**